### PR TITLE
Bindings for translation API

### PR DESCRIPTION
**Pull request by vidas3**

### CODE REVIEW

File: speech/translation_result.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		C.recognizer_result_handle_release(result.handle)
+		result.handle = C.SPXHANDLE_INVALID
+	}
+}
+
+func NewTranslationResultFromHandle(handle common.SPXHandle) (*TranslationResult, error) {
+	result := new(TranslationResult)
+	result.handle = uintptr2handle(handle)
+	var size C.size_t
+	ret := uintptr(C.translation_synthesis_result_get_audio_data(result.handle, nil, &size))
```
Comment (RytisInteractio) [inline speech/translation_result.go]: nil buffer intentional to get size and skip if 0?

Comment (vidas3) [reply speech/translation_result.go]: yes

File: speech/callback_helpers.go [hunk 2]
```diff
@@ -362,3 +362,28 @@ func synthesizerFireEventBookmarkReached(handle C.SPXRECOHANDLE, eventHandle C.S
...
+	defer mu.Unlock()
+	return translationSynthesisCallbacks[handle]
+}
+
+//export translatorFireEventSynthesizing
+func translatorFireEventSynthesizing(handle C.SPXRECOHANDLE, eventHandle C.SPXEVENTHANDLE) {
+	handler := getTranslationSynthesizingCallback(handle)
+	event, err := NewTranslationSynthesisEventArgsFromHandle(handle2uintptr(eventHandle))
+	if err != nil || handler == nil {
+		C.recognizer_event_handle_release(handle)
```
Comment (RytisInteractio) [inline speech/callback_helpers.go:385]: Why is only this branch releasing? Doesn't seem to get referenced anywhere in the function.

File: speech/speech_recognition_result.go [hunk 3]
```diff
@@ -72,6 +76,20 @@ func NewSpeechRecognitionResultFromHandle(handle common.SPXHandle) (*SpeechRecog
...
 	result.Text = C.GoString((*C.char)(buffer))
+	// Translation
+	var languageSize C.size_t
+	var textSize C.size_t
+	ret = uintptr(C.translation_text_result_get_translation(result.handle, 0, nil, nil, &languageSize, &textSize))
+	if ret != C.SPX_NOERROR {
+		return nil, common.NewCarbonError(ret)
+	}
+	language := C.malloc(C.sizeof_char * languageSize)
+	text := C.malloc(C.sizeof_char * textSize)
```
Comment (RytisInteractio) [inline speech/speech_recognition_result.go:91]: `language` `text` never freed

File: speech/callback_helpers.go (hunk 2 above)
Comment (vidas3) [reply speech/callback_helpers.go:385]: what do you mean? somebody has to release it?

Comment (RytisInteractio) [reply speech/callback_helpers.go:385]: `handle` is released if handler is nil or error, but doesn't seem to be referenced or released if they are not?

**Review by RytisInteractio — APPROVED** [review]

File: speech/callback_helpers.go (hunk 2 above)
Comment (vidas3) [reply speech/callback_helpers.go:385]: i think eventHandle should be released here?

Comment (RytisInteractio) [reply speech/callback_helpers.go:385]: Would make more sense. Would still need to be freed by the handler after handling it. Not sure who's supposed to be the owner - not finding any use of the function.

Comment (vidas3) [reply speech/callback_helpers.go:385]: event handle is released from consumer side

**Review by RytisInteractio — APPROVED** [review]

**Pull request merged**